application.go

```diff
@@ -109,6 +129,16 @@ type DeleteAppOpts struct {
 	Force bool `url:"force,omitempty"`
 }
 
+// SetIPAddressPerTask defines that the application will have a IP address defines by a external agent.
+// This configuration is not allowed to be used with Port or PortDefinitions.
+func (r *Application) SetIPAddressPerTask(ipAddressPerTask *IPAddressPerTask) *Application {
```

Comment:
I only meant the parameter as passed to the `SetIPAddressPerTask` method, not the field in the `Application` struct.

I'm basically in favor of making just the following change:

```go
func (r *Application) SetIPAddressPerTask(ipAddressPerTask IPAddressPerTask) *Application {
  r.Ports = make([]int, 0)
  r.EmptyPortDefinitions()
  r.IPAddressPerTask = &ipAddressPerTask
 
  return r
}
```

(Address operator gone in the method signature and "moved" to the assignment.)

It's just a tad easier for the user and indicates that we don't take `nil`s.

That I was also my point when [talking about discovery](https://github.com/gambol99/go-marathon/pull/223#discussion_r85009573). I suppose there was a misunderstanding, and that's why you de-pointerized the `Discovery` field in the `IPAddressPerTask` struct.